File: src/api/java/model/organization/entities/Pmf.java
package model.organization.entities;

import java.util.Collection;

import model.organization.Organization;
import model.organization.id.Identifiable;
import model.organization.relations.Assignment;

/**
 * The {@linkplain Pmf} interface defines the performance function entity of an {@linkplain Organization}.
 *
 * @author Christopher Zhong
 * @since 6.0
 */
public interface Pmf extends Identifiable<Pmf> {
	/**
	 * Computes the PMF for the given <code>Agent</code>, <code>Role</code>, and <code>Goal</code>.
	 *
	 * @param agent
	 *            the <code>Agent</code> to compute the PMF.
	 * @param role
	 *            the <code>Role</code> to compute the PMF.
	 * @param goal
	 *            the <code>Goal</code> to compute the PMF.
	 * @param assignments
	 *            the {@link Assignment} that may be used in computing the PMF.
	 * @return a <code>Double</code> value indicating the PMF value, or <code>null</code> if not applicable.
	 */
	Double pmf(Agent agent, Role role, InstanceGoal goal, Collection<Assignment> assignments);
}

File: chazm-api/src/main/kotlin/runtimemodels/chazm/api/entity/Pmf.kt
package runtimemodels.chazm.api.entity

import runtimemodels.chazm.api.id.Identifiable
import runtimemodels.chazm.api.id.PmfId
import runtimemodels.chazm.api.organization.Organization

/**
 * The [Pmf] interface defines the performance moderator function entity of an [Organization].
 *
 * @author Christopher Zhong
 * @since 6.0
 */
interface Pmf : Identifiable<Pmf, PmfId>
